### PR TITLE
Remove Hyphens

### PR DESCRIPTION
**Pull request by JavonTeo**
Remove unnecessary hyphens for windows PDF.

Additionally, fixed spacing issues due to link break: 
> "...effective for improving many natural languageprocessing tasks..." to "...effective for improving many natural language processing tasks..."

#37: Improve quality of texts extracted from PDF files

### CODE REVIEW

File: src/SciAssist/utils/windows_pdf2text.py [hunk 1]
```diff
@@ -19,7 +19,8 @@ def process_pdf(path):
     for page_layout in extract_pages(path):
         for element in page_layout:
             if "get_text" in dir(element):
-                text = element.get_text().replace("\n","")
+                text = element.get_text().replace("-\n","")
```
Comment (qolina) [inline src/SciAssist/utils/windows_pdf2text.py:22]: remove hyphen in the end of one line.

File: src/SciAssist/utils/windows_pdf2text.py [hunk 2]
```diff
@@ -19,7 +19,8 @@ def process_pdf(path):
...
+                text = text.replace("\n"," ")
```
Comment (qolina) [inline src/SciAssist/utils/windows_pdf2text.py:23]: concatenate breaking lines. But what if this is a normal paragraph break?

File: src/SciAssist/utils/windows_pdf2text.py (hunk 1 above)
Comment (JavonTeo) [reply src/SciAssist/utils/windows_pdf2text.py:22]: Yes the problem only appears for windows, not for linux.

File: src/SciAssist/utils/windows_pdf2text.py (hunk 2 above)
Comment (JavonTeo) [reply src/SciAssist/utils/windows_pdf2text.py:23]: If normal paragraph break, "\n" will be replaced by a space " ". Is this not the intended result?

File: src/SciAssist/utils/windows_pdf2text.py [hunk 3]
```diff
@@ -19,7 +19,8 @@ def process_pdf(path):
     for page_layout in extract_pages(path):
         for element in page_layout:
             if "get_text" in dir(element):
-                text = element.get_text().replace("\n","")
+                text = element.get_text().replace("-\n","") # Removes hyphen from a word split between lines
+                text = text.replace("\n"," ") # Replace "\n" with space (between words)
```
Comment (qolina) [inline src/SciAssist/utils/windows_pdf2text.py:23]: change back to the original code, designed to strip all spaces in the string.

Later, all segments/lines are joined with " ".

**Review by qolina — CHANGES_REQUESTED** [review]

**Review by qolina — APPROVED** [review]

File: src/SciAssist/utils/windows_pdf2text.py (hunk 2 above)
Comment (qolina) [reply src/SciAssist/utils/windows_pdf2text.py]: It is the desired.

**Pull request merged**